### PR TITLE
Support for opting out of global namespace pollution

### PR DESCRIPTION
**Pull request by tusharmath**

### CODE REVIEW

File: window.js [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+function defineParentNode (obj) {
+  obj.__defineGetter__('parentNode', function () { return this.parentElement });
+}
+
+defineParentNode(Element.prototype)
+defineParentNode(Comment.prototype)
+defineParentNode(Text.prototype)
+defineParentNode(Node.prototype)
+
+module.exports = { Document, Node, Element, Comment, Text, document: new Document() };
```
Comment (tdumitrescu) [inline window.js]: NB this object literal shorthand means you'll need node.js v4 or above to use `html-element`. I personally don't see a problem with it but I don't know how @1N50MN14 feels about backwards compatibility with, e.g. v0.12. If you do want to deprecate support for old versions, I'd suggest indicating the requirements in `package.json` with the `engines` field, and a major version bump for `html-element`.

File: index.js [hunk 2]
```diff
@@ -1,376 +1,7 @@
+const {Document, Node, Element, Comment, Text, document} = require('./window')
```
Comment (tdumitrescu) [inline index.js]: likewise here re: object literal shorthand

**Comment by tdumitrescu** [comment]
Aside from the minor syntax comment, the implementation here looks OK to me. But has anyone considered that the opposite API might be more sensible, namely, that importing this library by default (from `index.js`) should not patch the global state, and an optional module would do it. E.g. the lib's consumers could do either
```js
import { document, Element } from 'html-element';
```
for no side effects, or else
```js
import 'html-element/global-shims';
```
to have everything automatically patched globally. Any thoughts?

**Comment by 1N50MN14** [comment]
Yup I think  `import 'html-element/global-shims';`makes more sense also in regards to hyperscript

**Comment by tusharmath** [comment]
@1N50MN14 If by default `global-shims` are not loaded, it would completely break hyperscript and as you said it should not be a breaking change.

**Comment by tdumitrescu** [comment]
If it comes with a major version bump (and updated documentation) then it
shouldn't break anything, unless the hyperscript package left the
html-element version completely open with > or >=.

**Comment by tdumitrescu** [comment]
Looks like as of right now Hyperscript has `html-element` at `^1.4.0` (https://github.com/dominictarr/hyperscript/blob/68706db8c9dc6cbb1497f2a29f66d08e9d549277/package.json#L20), so a bump to 2.0.0 wouldn't get picked up until someone explicitly updates hyperscript to use the new version.

**Comment by tusharmath** [comment]
@tdumitrescu @1N50MN14 
can you take a look at the implementation here? I will squash the comments if everything is fine. 

Can you please help me with the documentation too?

**Comment by tdumitrescu** [comment]
Looks good, much more limited changeset than before. I can add some documentation updates when merging.

**Comment by tusharmath** [comment]
squashed the commits.

**Comment by tdumitrescu** [comment]
Merged in https://github.com/1N50MN14/html-element/commit/ca3d87b3701f438f9ddbb05c669fac3a1a14bfd1. Thanks!

**Pull request closed without merging**